### PR TITLE
Added 2d overlay functions to simplify drawing

### LINKED ISSUE #973: Split 2D overlay from Adaptive Formation into it own function
See [here](https://github.com/RoboJackets/robocup-software/pull/958#discussion_r112817441)

### PR DESCRIPTION
**Pull request by JNeiger**
Closes #973. I decided to ditch the play class function way. This is a lot simpler to implement and deal with I think

### CODE REVIEW

**Comment by JNeiger** [comment]
@jgkamat I might have broken circleci

**Comment by jgkamat** [comment]
There was a missing `__init__` file in the new module you added, I just patched it. It should be good to go now!

**Comment by JNeiger** [comment]
Is there a problem with the style too? The patch fails with the error `fatal: unrecognized input`

**Comment by jgkamat** [comment]
Uh thats actually a mistake on my part, thanks for catching that. I'll fix that, don't worry about it for now.

**Comment by JNeiger** [comment]
This is ready for a review whenever someone gets a chance

**Comment by jgkamat** [comment]
I'm going to be super busy for the next couple of weeks working on a few things, can @chachmu or @joshhting take this? :)

**Comment by joshhting** [comment]
Can you explain what the color gradient means here?

**Comment by JNeiger** [comment]
I'm not really sure what you mean by that, but this is the results of a scoring algorithm based on field position. 
image 
The idea is to visualize the score of some scoring algorithm at each point in the field where orange is a higher score and blue is a lower score. We use the scoring for adaptive formation (and probably the new defense eventually too)

File: soccer/gameplay/plays/testing/test_adaptive_formation_weights.py [hunk 1]
```diff
@@ -29,91 +30,24 @@ def __init__(self):
...
-        x_cent = max_pt.x
-        y_cent = max_pt.y
+                sublist.append(evaluation.field.field_pos_coeff_at_pos(
+                    pt, 0.1, .2, 0.02))
+                #vals.append(1-evaluation.field.space_coeff_at_pos(pt))
+                #vals.append(evaluation.shooting.eval_shot(pt))
+                #vals.append(evaluation.defensive_positioning.estimate_risk_score(pt))
+                #vals.append(5 * evaluation.passing_positioning.eval_single_point(
+                #    main.ball().pos, main.our_robots(), (0.01, 3, 0.02),
+                #    (2, 2, 15, 1), pt.x, pt.y))
```
Comment (joshhting) [inline soccer/gameplay/plays/testing/test_adaptive_formation_weights.py]: These commented out lines should also be changed for the 2d array

**Pull request merged**